### DISCUSSION TITLE
Http query example

### DISCUSSION THREAD

**Discussion by rajrahul** [geldata/gel#3492] [Q&A] [answered]
Could you point me to a complete HTTP query sample? Could not get it working from the documentation at https://www.edgedb.com/docs/clients/90_edgeql/index#edgeql-over-http

The port is open and I could connect using telnet.

**Comment by elprans** [comment, marked as answer]
The API is documented [on the next page](https://www.edgedb.com/docs/clients/90_edgeql/protocol):

```
curl -X POST  http://127.0.0.1:<instance-port>/db/<database-name>/edgeql \
  -H 'Content-Type: application/json' \
  -d '{"query": "...", "variables": { "varName": "varValue", ... }}'
```

**Reply by rajrahul** [reply]
Here is the credential file:
cat /home/rr/.config/edgedb/credentials/quickstart.json 
{
  "port": 10701,
  "user": "edgedb",
  "password": "5AMCkMriuTAy1wXhS06Ie5dj",
  "database": "edgedb",
...

I have created some databases using the CLI and listing displays
List of databases :
  edgedb
  dracula
  apex

When I curl:

curl -X POST  http://127.0.0.1:10701/db/apex/edgeql \
  -H 'Content-Type: application/json' \
  -d '{"query": "select Address{postal_code}", "variables": {}}'

It always returns the error "unknown path". Using the database 'edgedb' also results in the same error. No further error info is available.

**Reply by elprans** [reply]
Do you have `using extension edgeql_http;` in your schema?

**Reply by rajrahul** [reply]
I created the tables from CLI using a 'start migration' as shown below. Not sure where to add that , CLI does not accept that statement as well.
START MIGRATION TO {
  module default {
    ....
  }    
};

POPULATE MIGRATION;
COMMIT MIGRATION;

**Reply by rajrahul** [reply]
I created a new schema enabling "using extension edgeql_http;". Still the same error persist

**Reply by rajrahul** [reply]
It works, I had given a wrong port. Thanks for the inputs.
